CMakeLists.txt

```diff
@@ -9,38 +9,55 @@ include(ExternalProject)
 
 file(MAKE_DIRECTORY ${CATKIN_DEVEL_PREFIX}/include)
 
+SET(CONTRIB_NAME "opencv3_contrib")
+ExternalProject_Add(${CONTRIB_NAME}
+  URL https://github.com/opencv/opencv_contrib/archive/3.1.0.zip
+  UPDATE_COMMAND ""
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ""
+)
+
 ExternalProject_Add(opencv3_src
-  URL https://github.com/Itseez/opencv/archive/3.0.0-beta.zip
+  URL https://github.com/Itseez/opencv/archive/3.1.0.zip
   UPDATE_COMMAND ""
-  PATCH_COMMAND cd ../opencv3_src/
   CONFIGURE_COMMAND cd ../opencv3_src && cmake
-    -D WITH_QT=ON
-    -D WITH_OPENGL=ON
-    -D WITH_VTK=ON
-    -D WITH_OPENMP=ON
```

Comment:
I actually copied the whole project from `opencv2_catkin` and changed the url and added the extra contrib package.
(same for all comments below)